aws/resource_aws_api_gateway_stage.go

```diff
@@ -18,6 +19,9 @@ func resourceAwsApiGatewayStage() *schema.Resource {
 		Read:   resourceAwsApiGatewayStageRead,
 		Update: resourceAwsApiGatewayStageUpdate,
 		Delete: resourceAwsApiGatewayStageDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsApiGatewayStageImportState,
```

Comment:
To support import when the read function is not using `d.Id()` directly, we just need to simply `d.Set()` the attributes required for read. I would also recommend splitting on `/` which matches the format of the execution ARN. Something like this should work:

```go
State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
	idParts := strings.Split(d.Id(), "/")
	if len(idParts) != 2 || len(d.Id()) < 3 {
		return []*schema.ResourceData{}, fmt.Errorf("Wrong format of ID: %s. Please follow 'rest-api-id/stage-name'", d.Id())
	}
	d.Set("rest_api_id", idParts[0])
	d.Set("stage_name", idParts[1])
	return []*schema.ResourceData{d}, nil
},
```